### PR TITLE
Installation and Setup Instructions for IntelliJ Idea and VSCode

### PR DESCRIPTION
**Pull request by TheReconPilot**
- Wrote a brief intro on the `index.rst` file.
- Added installation and setup instructions in `setup.rst`.
- Put images in `source/_static/images`

### CODE REVIEW

**Comment by TheReconPilot** [comment]
Added some content on command line.

**Review by dpcherian — CHANGES_REQUESTED** [review]
The actual BharatSim source code directory will probably just be called ``BharatSim``, and not ``BharatSim-master`` which was just a name of the downloaded repository we sent you guys. Could you change this document to reflect that?

We will include a dummy data file with the repository, so you don't need to include how to generate the dummy data here. Just work under the assumption that an appropriately named file is placed in the correct place. (You can keep the error message, but put in an explanation saying that a file called ``citizen10k.csv` should exist in the appropriate folder, in case someone deletes it by accident.) It is an important thing to describe, just not here. I would suggest putting that in a separate section, I've added a section under Miscellaneous in the Google Sheet.

**Comment by TheReconPilot** [comment]
Ah okay. Will do the changes and commit soon.

**Comment by TheReconPilot** [comment]
Made the changes

**Review by dpcherian — COMMENTED** [review]
Overall looks great, good work. Mentioned some changes that might be good to implement in the images. See if you can add them?

File: source/setup.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  
+* Now, we **run the SIR Model**. SIR is a simple compartmental model to analyze epidemics, where a person can be either Susceptible (S), Infected (I) or Recovered (R). We will see SIR Model in detail in the Epidemiology section. 
+
+  * Do ``sbt run`` and select the class number associated to ``com.bharatsim.examples.epidemiology.sir.Main``. It should start running the simulation. It may appear as if the class number is not being typed, but it is! Just input the number and press ENTER.
+
+    .. error:: If it gives an error like ``Ingestion Failed : java.nio.file.NoSuchFileException: citizen10k.csv``, make sure there is a file named ``citizen10k.csv`` inside the BharatSim folder. If it is not present, it might have been mistakenly deleted or misplaced. Get the source code again in that case.
+
+    It should look like this:
+
+    .. image:: _static/images/cli-sir-run.png
```
[the remark this answers is not in the document]
Comment (TheReconPilot) [reply source/setup.rst:141]: Fixed in 770fc28

File: source/setup.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+* The output CSV file is present at ``BharatSim/src/main/resources/output_unixtimestamp.csv``. This contains the output as per the specification in the program. This can be used to further analyze the results of the SIR Model run.
+
+This is how Scala programs can be run through the command line.
+
+.. tip:: Simply typing ``sbt`` will run the sbt console. The other commands can now be run in succession simply as ``compile``, ``run`` and more.
+
+  .. image:: _static/images/sbt-console.png
```
[the remark this answers is not in the document]
Comment (TheReconPilot) [reply source/setup.rst:149]: Fixed in 770fc28

File: source/setup.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  
+  * In IntelliJ Idea, on the left is the project directory structure and tree. Navigate to ``BharatSim/src/main/scala/com/bharatsim/examples/epidemiology/sir`` and open the ``Main.scala`` file. 
+  
+  * There will again be a Green arrow/triangle besides the line containing ``object Main``. Click on it, and ``Run 'Main'``.
+  
+    .. error:: If it gives an error like ``Ingestion Failed : java.nio.file.NoSuchFileException: citizen10k.csv``, make sure there is a file named ``citizen10k.csv`` inside the BharatSim folder. If it is not present, it might have been mistakenly deleted or misplaced. Get the source code again in that case.
+
+  * Wait till the program finishes running. At the end, it should look like this:
+
+    .. image:: _static/images/sir-run.png
```
[the remark this answers is not in the document]
Comment (TheReconPilot) [reply source/setup.rst:78]: Yeah that was a particularly bad screenshot. I switched to a better tool now. 

Fixed in aac8b5a

File: source/setup.rst [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+* Now, we **run the SIR Model**. SIR is a simple compartmental model to analyze epidemics, where a person can be either Susceptible (S), Infected (I) or Recovered (R). We will see SIR Model in detail in the Epidemiology section.
+  
+  * In VSCode, on the left is the project directory structure and tree. Navigate to ``BharatSim/src/main/scala/com/bharatsim/examples/epidemiology/sir`` and open the ``Main.scala`` file. 
+  * There will again be a ``run | debug`` above the line containing ``object Main``. Click on ``run``.
+  
+    .. error:: If it gives an error like ``Ingestion Failed : java.nio.file.NoSuchFileException: citizen10k.csv``, make sure there is a file named ``citizen10k.csv`` inside the BharatSim folder. If it is not present, it might have been mistakenly deleted or misplaced. Get the source code again in that case.
+
+  * Wait till the program finishes running. At the end, it should look like this:
+
+    .. image:: _static/images/vscode-sir-run.png
```
[the remark this answers is not in the document]
Comment (TheReconPilot) [reply source/setup.rst:114]: Fixed in aac8b5a

File: source/setup.rst [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+IntelliJ Idea
+~~~~~~~~~~~~~
+
+* Download the `IntelliJ Idea Community Edition <https://www.jetbrains.com/idea/download/>`_. 
+
+
+.. note:: On Linux, a ``.tar.gz`` file is downloaded. 
+
+  * Unzip the archive: ``tar xvf archivename.tar.gz``.
```
[the remark this answers is not in the document]
Comment (TheReconPilot) [reply source/setup.rst]: Done on 4ef5821

**Review by dpcherian — APPROVED** [review]
Looks great! Thanks for making the changes, I'm approving the pull-request.

**Pull request merged**